src/setfit/integrations.py

```diff
@@ -1,28 +1,51 @@
 import importlib.util
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Optional
 
 from .utils import BestRun
+from .trainer import SetFitTrainer
 
 
-if TYPE_CHECKING:
-    from .trainer import SetFitTrainer
+def is_optuna_available() -> bool:
+    """Check if Optuna is available."""
+    return importlib.util.find_spec("optuna") is not None
 
 
-def is_optuna_available():
-    return importlib.util.find_spec("optuna") is not None
+def default_hp_search_backend() -> Optional[str]:
+    """Get the default backend for hyperparameter search."""
+    return "optuna" if is_optuna_available() else None
+
+
+def run_hp_search_optuna(
+    trainer: SetFitTrainer, 
+    n_trials: int, 
+    direction: str, 
+    timeout: Optional[int] = None,
+    n_jobs: int = 1,
+    **kwargs,
+) -> BestRun:
+    """Run hyperparameter search with Optuna.
 
+    Args:
+        trainer: SetFitTrainer object to train and evaluate the model.
+        n_trials: Number of trials to run for hyperparameter search.
+        direction: Optimization direction for the study.
+        timeout: Maximum duration in seconds for running the trials.
+        n_jobs: Number of parallel jobs to run.
+        **kwargs: Additional arguments to pass to `optuna.create_study`.
 
-def default_hp_search_backend():
-    if is_optuna_available():
-        return "optuna"
+    Returns:
+        A BestRun object containing the best hyperparameters and results.
 
+    Raises:
+        ImportError: If Optuna is not installed.
+    """
+    if not is_optuna_available():
+        raise ImportError("Optuna is not available. Please install it to run hyperparameter search with Optuna.")
 
-def run_hp_search_optuna(trainer: "SetFitTrainer", n_trials: int, direction: str, **kwargs) -> BestRun:
     import optuna
 
-    # Heavily inspired by transformers.integrations.run_hp_search_optuna
-    # https://github.com/huggingface/transformers/blob/cbb8a37929c3860210f95c9ec99b8b84b8cf57a1/src/transformers/integrations.py#L160
```

Comment:
If possible, I'd like to keep these comments. They might be useful for other collaborators.